share/picongpu/tests/PusherScaling/lib/python/test/Pusher/MainTest.py

```diff
@@ -0,0 +1,285 @@
+        x/y_poss (np.array): Relative in-Cell positions of the particle in the
+                             series
+        x/y_offSet (np.array): All node-positions of the particle in the series
+        period (int): openPMD period of the simulation
+        charge (float): Charge of particle
+        mass (float): Mass of particle
+        x/y_momentum (float): All momenta of the particle in the series
+        R_c (float): Radius of gyration of the particle at timestep 0
+
+        """
+
+        num_iter = len(self.series.iterations)
+
+        # create arrays for saving the data
+
+        # in-cell position
+        x_poss = np.zeros(num_iter)
+        y_poss = np.zeros(num_iter)
+
+        # cells
+        x_offSet = np.zeros(num_iter)
+        y_offSet = np.zeros(num_iter)
+
+        # momentum
+        x_momentum = np.zeros(num_iter)
+        y_momentum = np.zeros(num_iter)
+
+        i = self.series.iterations[0]
+        # charge
+        self.charge = i.particles["e"]["charge"][io.Mesh_Record_Component.SCALAR]
+        # mass
+        self.mass = i.particles["e"]["mass"][io.Mesh_Record_Component.SCALAR]
+
+        # calculating the openPMD-period for later use
+        indices = np.array(self.series.iterations)
+        period = indices[1] - indices[0]  # openPMD-period
+
+        for iteration in indices:
+            i = self.series.iterations[iteration]  # current position
+
+            # node coordinates
+            pos2_x = i.particles["e"]["position"]["x"].load_chunk()
+            pos2_y = i.particles["e"]["position"]["y"].load_chunk()
+
+            # InCell coordinates
+            pos_off2_x = i.particles["e"]["positionOffset"]["x"].load_chunk()
+            pos_off2_y = i.particles["e"]["positionOffset"]["y"].load_chunk()
+
+            p_x = i.particles["e"]["momentum"]["x"].load_chunk()
+            p_y = i.particles["e"]["momentum"]["y"].load_chunk()
+
+            self.series.flush()
+
+            array_index = int(iteration / period)
+            # save the read data in arrays for later usage
+            x_poss[array_index] = pos2_x
+            y_poss[array_index] = pos2_y
+
+            x_offSet[array_index] = pos_off2_x
+            y_offSet[array_index] = pos_off2_y
+
+            x_momentum[array_index] = p_x
+            y_momentum[array_index] = p_y
+
+        # all radii in the series
+        radius = (
+            np.sqrt(x_momentum**2 + y_momentum**2)
+            * self.params["unit_mass"]
+            * self.params["unit_speed"]
+            / (abs(self.charge[0]) * self.params["unit_charge"] * self.B)
+            / self.params["cell_depth"]
+            / self.params["unit_length"]
+        )
+
+        R_c = radius[0]  # original radius
+
+        # half_step
+        steps_per_turn = 0.825e-12 / (self.params["dt"] * self.params["unit_time"])
+        halfStepPhase = np.pi / steps_per_turn
+
+        # coordinate transformation to reposition the origin at the center of
```

Comment:
Whenever you've got a comment explaining what the next few lines do, extract those lines into a function and give them a name and a docstring.

Reply:
done as in the other test.